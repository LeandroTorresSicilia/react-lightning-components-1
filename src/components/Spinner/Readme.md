Spinner expamples:

    <div>
        <div className="slds-p-bottom--medium slds-is-relative" style={{ height: '5rem', width: '4rem' }} >
            <Spinner />
        </div>
        <div className="slds-p-bottom--medium slds-is-relative" style={{ height: '6rem', width: '4rem' }}>
            <Spinner variant="brand" size="large" />
        </div>
        <div className="slds-p-bottom--medium slds-is-relative" style={{ height: '4rem', width: '4rem', backgroundColor: 'green', borderRadius: '0.4rem' }}>
            <Spinner variant="inverse" size="small" />
        </div>
    </div>