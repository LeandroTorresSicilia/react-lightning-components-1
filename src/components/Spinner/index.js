/* eslint-disable react/require-default-props */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

export default class Spinner extends Component {
    getContainerClass() {
        const { size, variant } = this.props;

        return classnames('slds-spinner', {
            'slds-spinner_brand': variant === 'brand',
            'slds-spinner_inverse': variant === 'inverse',
            'slds-spinner_large': size === 'large',
            'slds-spinner_medium': size === 'medium',
            'slds-spinner_small': size === 'small',
            'slds-spinner_x-small': size === 'x-small',
            'slds-spinner_xx-small': size === 'xx-small',
        });
    }

    renderAssistiveText() {
        const { assistiveText } = this.props;

        if (assistiveText) {
            return (
                <span className="slds-assistive-text">
                    {assistiveText}
                </span>
            );
        }

        return null;
    }

    renderSpinner() {
        return (
            <div className={this.getContainerClass()} role="status">
                {this.renderAssistiveText()}
                <div className="slds-spinner__dot-a" />
                <div className="slds-spinner__dot-b" />
            </div>
        );
    }

    render() {
        return this.props.isVisible ? this.renderSpinner() : null;
    }
}

Spinner.propTypes = {
    /** Description for assistive sreen readers */
    assistiveText: PropTypes.string,
    /** Show/Hide the spinner */
    isVisible: PropTypes.bool,
    /** The size of the spinner */
    size: PropTypes.oneOf(['large', 'medium', 'small', 'x-small', 'xx-small']),
    /** The variant of the spinner */
    variant: PropTypes.oneOf(['default', 'brand', 'inverse']),
};

Spinner.defaultProps = {
    isVisible: true,
    size: 'medium',
    variant: 'default',
};
