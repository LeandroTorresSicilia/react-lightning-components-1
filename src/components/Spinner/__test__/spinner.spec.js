import React from 'react';
import renderer from 'react-test-renderer';
import Spinner from './../index';

describe('<Spinner/>', () => {
    it('should render the right markup for brand variant', () => {
        const component = renderer.create(<Spinner variant="brand" />);

        expect(component).toMatchSnapshot();
    });

    it('should render the spinner by default if isVisible is not passed', () => {
        const component = renderer.create(<Spinner />);

        expect(component).toMatchSnapshot();
    });

    it('should render nothing if isVisible is set to false', () => {
        const component = renderer.create(<Spinner isVisible={false} />);

        expect(component).toMatchSnapshot();
    });

    it('should render the assistiveText if it is passed', () => {
        const component = renderer.create(<Spinner assistiveText="Loading" />);

        expect(component).toMatchSnapshot();
    });
});
