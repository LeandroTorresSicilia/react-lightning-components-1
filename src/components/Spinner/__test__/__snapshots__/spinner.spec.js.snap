// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner/> should render nothing if isVisible is set to false 1`] = `null`;

exports[`<Spinner/> should render the assistiveText if it is passed 1`] = `
<div
  className="slds-spinner slds-spinner_medium"
  role="status"
>
  <span
    className="slds-assistive-text"
  >
    Loading
  </span>
  <div
    className="slds-spinner__dot-a"
  />
  <div
    className="slds-spinner__dot-b"
  />
</div>
`;

exports[`<Spinner/> should render the right markup for brand variant 1`] = `
<div
  className="slds-spinner slds-spinner_brand slds-spinner_medium"
  role="status"
>
  <div
    className="slds-spinner__dot-a"
  />
  <div
    className="slds-spinner__dot-b"
  />
</div>
`;

exports[`<Spinner/> should render the spinner by default if isVisible is not passed 1`] = `
<div
  className="slds-spinner slds-spinner_medium"
  role="status"
>
  <div
    className="slds-spinner__dot-a"
  />
  <div
    className="slds-spinner__dot-b"
  />
</div>
`;
